### PR TITLE
Move tests to invoke `te.exe` directly instead of using VSTest runner

### PR DESCRIPTION
**Pull request by miniksa**
Moves the tests from using the `vstest.console.exe` route to just using `te.exe`.

PROs:
- `te.exe` is significantly faster for running tests because the TAEF/VSTest adapter isn't great.
- Running through `te.exe` is closer to what our developers are doing on their dev boxes
- `te.exe` is how they run in the Windows gates.
- `te.exe` doesn't seem to have the sporadic `0x6` error code thrown during the tests where somehow the console handles get lost
- `te.exe` doesn't seem to repro the other intermittent issues that we have been having that are inscrutable. 
- Fewer processes in the tree (te is running anyway under `vstest.console.exe`, just indirected a lot
- The log outputs scroll live with all our logging messages instead of suppressing everything until there's a failure
- The log output is actually in the order things are happening versus vstest.

CONs:
- No more code coverage.
- No more test records in the ADO build/test panel.
- Tests really won't work inside Visual Studio at all.
- The log files are really big now
- Testing is not a test task anymore, just another script.

Refuting each CON:
- We didn't read the code coverage numbers
- We didn't look at the ADO test panel results or build-over-build velocities
- Tests didn't really work inside Visual Studio anyway unless you did the right incantations under the full moon.
- We could tone down the logging if we wanted at either the te.exe execution time (with a switch) or by declaring properties in the tests/classes/modules that are very verbose to not log unless it fails.
- I don't think anyone cares how they get run as long as they do.

### CODE REVIEW

**Comment by miniksa** [comment]
K. I know this is bad. I need to use a var to determine the config/platform and/or a PowerShell script that can replace the pattern ** search from what the VS test task could do.

**Comment by miniksa** [comment]
`vstest.console.exe` runtime x86 unit tests: 5m8s
`te.exe` runtime x86 unit tests: 46s

Yeah, uh, let's just run them direct.

**Comment by DHowett-MSFT** [comment]
```yaml
# EARLIER

- task: PowerShell@2
  displayName: 'Rationalize build platform'
  inputs:
    targetType: inline
    script: |
      $Arch = "$(BuildPlatform)"
      If ($Arch -Eq "x86") { $Arch = "Win32" }
      Write-Host "##vso[task.setvariable variable=RationalizedBuildPlatform]${Arch}"

# LATER

- task: PowerShell@2
  displayName: 'Run Unit Tests'
  inputs:
    targetType: inline
    script: |
      $TestRoot = ".\\bin\\$(RationalizedBuildPlatform)\\$(BuildConfiguration)"
      & ${TestRoot}\TE.exe ${TestRoot}\*.Unit.Test*.dll
  condition: bla bla bla
```

perhaps? Maybe?

**Comment by miniksa** [comment]
feature tests `vstest.console.exe`: 2m42s
feature tests `te.exe`: 1m39s

**Comment by miniksa** [comment]
> ```yaml
> # EARLIER
> 
> - task: PowerShell@2
>   displayName: 'Rationalize build platform'
>   inputs:
>     targetType: inline
>     script: |
>       $Arch = "$(BuildPlatform)"
>       If ($Arch -Eq "x86") { $Arch = "Win32" }
>       Write-Host "##vso[task.setvariable variable=RationalizedBuildPlatform]${Arch}"
> 
> # LATER
> 
> - task: PowerShell@2
>   displayName: 'Run Unit Tests'
>   inputs:
>     targetType: inline
>     script: |
>       $TestRoot = ".\bin\$(RationalizedBuildPlatform)\$(BuildConfiguration)"
>       & ${TestRoot}\TE.exe ${TestRoot}\*.Unit.Test*.dll
>   condition: bla bla bla
> ```
> 
> perhaps? Maybe?

Yeah that looks a bit better than what I came up with.

**Comment by oising** [comment]
ERMAGAHD

File: src/foo.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+asdf
```
Comment (DHowett-MSFT) [inline src/foo.txt]: oh i get it: this file makes the CI run

Comment (miniksa) [reply src/foo.txt]: hush you it's still a draft.

**Comment by DHowett-MSFT** [comment]
nit: change name to be imperative (Move, Do, Fix, Teach, etc.)

**Comment by zadjii-msft** [comment]
image
[heres the build](https://dev.azure.com/ms/Terminal/_build/results?buildId=62155&view=logs&jobId=e7365288-57d7-5349-8dfa-9c195127bd77&j=e7365288-57d7-5349-8dfa-9c195127bd77&t=baf96a51-7e9f-549a-4d5d-6ccfbe0186d8)

I'm inclined to sign off anyways for the speed boost, but it looks like the roundtrip tests _still_ hang in just TAEF on CI 😢

**Comment by DHowett-MSFT** [comment]
At least we can now see which one was running when it hangs! When DevOps actually lets you scroll continuously (instead of jumping from line 110000 to line 100025...)

**Comment by DHowett-MSFT** [comment]
```
2020-02-06T23:53:14.6908403Z StartGroup: ConptyOutputTests::WriteAFewSimpleLines
2020-02-06T23:53:14.6911125Z Write more lines of output. We should use 
2020-02-06T23:53:14.6911809Z  to move the cursor
2020-02-06T23:53:14.6912996Z Verify: IsNotNull(_pVtRenderEngine.get())
2020-02-06T23:53:14.6914899Z Verify: IsGreaterThan(expectedOutput.size(), static_cast<size_t>(0)): writing="ï¿½[2J", expecting 4 strings
2020-02-06T23:53:14.6915910Z Expected =	"ï¿½[2J"
2020-02-06T23:53:14.6917308Z Actual =	"ï¿½[2J"
```

**Comment by miniksa** [comment]
Yes, I will go dig into that test now and fixing it or turning it off (with a TODO) may be a part of this too.

File: src/host/ut_host/VtIoTests.cpp [hunk 2]
```diff
@@ -360,7 +362,7 @@ void VtIoTests::RendererDtorAndThread()
         // EnablePainting gets called, and if that happens, then the thread will
         // never get destructed. This will only ever happen in the vstest test runner,
         // which is what CI uses.
-        Sleep(500);
+        /*Sleep(500);*/
```
Comment (carlos-zamora) [inline src/host/ut_host/VtIoTests.cpp]: stray comment. Delete?
(and below)

**Comment by miniksa** [comment]
This build is running 1767 unit tests and 405 feature tests (another PR): https://dev.azure.com/ms/Terminal/_build/results?buildId=62323&view=logs&j=0f660c0a-4423-5c5d-276b-f98b39c69e13&t=ecf341a7-dafd-5ce6-eabd-86286d881b78

This build that just ran has 1753 unit tests and 405 feature test: https://dev.azure.com/ms/Terminal/_build/results?buildId=62403&view=logs&j=0f660c0a-4423-5c5d-276b-f98b39c69e13&t=f0309e56-3cac-5248-a471-a2958657f90a

I think I'm missing one module...

**Comment by miniksa** [comment]
Yeah I'm missing TerminalAppUnitTests because they're in a bin subdirectory...

**Comment by miniksa** [comment]
OK it blended mostly. Take out the failboat test and update the solution. If this works, we're good to go.

File: src/host/ut_host/VtIoTests.cpp (hunk 2 above)
Comment (miniksa) [reply src/host/ut_host/VtIoTests.cpp]: Gonna leave it for now because these tests are notorious.

**Comment by DHowett-MSFT** [comment]
Get them good-good signoffs!

**Pull request merged**